src/Payments.sol

```diff
@@ -861,32 +824,25 @@ contract Payments is
         Rail storage rail = rails[railId];
         Account storage payer = accounts[rail.token][rail.from];
 
-        // Update the payer's lockup to account for elapsed time
-        settleAccountLockup(payer);
-
-        // Handle terminated rails
-        if (isRailTerminated(rail)) {
-            uint256 maxTerminatedRailSettlementEpoch = maxSettlementEpochForTerminatedRail(
-                    rail
-                );
-
-            // If rail is already fully settled but still active, finalize it
-            if (rail.settledUpTo >= maxTerminatedRailSettlementEpoch) {
-                finalizeTerminatedRail(rail, payer);
-                return (
-                    0,
-                    rail.settledUpTo,
-                    "rail fully settled and finalized"
-                );
-            }
-
-            // For terminated but not fully settled rails, limit settlement window
-            untilEpoch = min(untilEpoch, maxTerminatedRailSettlementEpoch);
+        // Handle terminated and fully settled rails that are still not finalised
+        if (
+            isRailTerminated(rail) &&
+            rail.settledUpTo >= maxSettlementEpochForTerminatedRail(rail)
+        ) {
+            finalizeTerminatedRail(rail, payer);
+            return (0, rail.settledUpTo, "rail fully settled and finalized");
         }
 
-        uint256 maxLockupSettlementEpoch = payer.lockupLastSettledAt +
-            rail.lockupPeriod;
-        uint256 maxSettlementEpoch = min(untilEpoch, maxLockupSettlementEpoch);
+        // Calculate the maximum settlement epoch based on account lockup
+        uint256 maxSettlementEpoch;
+        if (!isRailTerminated(rail)) {
+            maxSettlementEpoch = min(untilEpoch, payer.lockupLastSettledAt);
```

Comment:
I need to stop working on FVM tech debt and take some time tomorrow to try to actually fix the underlying issue here. The fact that we can't settle these locked funds is really annoying me, there's no technical reason this isn't possible.

Reply:
(and maybe I'm wrong, but this is a case where I need to prove this to myself)

Reply:
Sure. I have thought a lot about this, have written tests to understand this better and doing this really simplifies accounting without sacrificing much in the way of UX. 

I am sure we can come up with **_more_** arithmetic to be able to do this but it adds one more dimension of complexity to an already complex contract and I would like to first give this contract to people and complain about not being able to get locked funds without rail termination before we introduce this complexity.